File: app/src/main/java/com/meetferrytan/mvpdaggerstore/util/interceptors/ConnectivityInterceptor.java
package com.meetferrytan.mvpdaggerstore.util.interceptors;

/**
 * Created by ferrytan on 9/28/17.
 */

import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import java.io.IOException;
import java.net.ConnectException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class ConnectivityInterceptor implements Interceptor {
    public static final String NO_INTERNET_CONNECTION_MESSAGE = "No Connectivity Exception";
    private final ConnectivityManager mConnectivityManager;

    public ConnectivityInterceptor(ConnectivityManager connectivityManager) {
        mConnectivityManager = connectivityManager;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        if (!isConnected()) {
            throw new ConnectException("Not connected to network");
        }

        Request.Builder builder = chain.request().newBuilder();
        return chain.proceed(builder.build());
    }

    private boolean isConnected() {
        NetworkInfo netInfo = mConnectivityManager.getActiveNetworkInfo();
        return (netInfo != null && netInfo.isConnected());
    }

}

File: app/src/main/java/com/meetferrytan/mvpdaggerstore/util/interceptors/ConnectivityInterceptor.kt
package com.meetferrytan.mvpdaggerstore.util.interceptors

/**
 * Created by ferrytan on 9/28/17.
 */

import android.net.ConnectivityManager
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import java.net.ConnectException

class ConnectivityInterceptor(private val mConnectivityManager: ConnectivityManager) : Interceptor {

    private val isConnected: Boolean
        get() {
            val netInfo = mConnectivityManager.activeNetworkInfo
            return netInfo != null && netInfo.isConnected
        }

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        if (!isConnected) {
            throw ConnectException(NO_INTERNET_CONNECTION_MESSAGE)
        }

        val builder = chain.request().newBuilder()
        return chain.proceed(builder.build())
    }

    companion object {
        val NO_INTERNET_CONNECTION_MESSAGE = "Not connected to network"
    }

}